src/cml.js

```diff
@@ -159,9 +180,70 @@ class CML {
       ? ''
       : '![CML watermark](https://raw.githubusercontent.com/iterative/cml/master/assets/watermark.svg)';
 
-    const report = `${userReport}\n\n${watermark}`;
+    let userReport = testReport;
+    try {
+      if (!userReport) {
+        userReport = await fs.promises.readFile(markdownFile, 'utf-8');
+      }
+    } catch (err) {
+      if (!watch) throw err;
+    }
+
+    let report = `${userReport}\n\n${watermark}`;
     const drv = getDriver(this);
 
+    const publishLocalFiles = async (tree) => {
+      const nodes = [];
+
+      visit(tree, ['definition', 'image', 'link'], (node) => nodes.push(node));
+
+      const visitor = async (node) => {
+        if (node.url && node.alt !== 'CML watermark') {
```

Comment:
elsewhere we check only for a string `watermark.svg`, not `node.alt == 'CML watermark'`.

Can we haz consistency? Not fussy about which is better.

https://github.com/iterative/cml/blob/3f2aadcb10a995b57f37c82c7daa9eaae4160376/src/cml.js#L168

Reply:
It's not about consistency, it's about intent:

1. `watermark.svg` is used to check if a comment was created by CML
2. `node.alt` is used to check if the current Markdown image is the CML watermark

They are not equivalent; the second is used only when traversing the Markdown AST.

Reply:
yes exactly, they're incompatible? The following will cause undefined behaviour, right?

1. `echo '![pwned](https://examplewatermark.svg)' >> report.md`
2. `echo '![CML watermark](https://pwn.ed)' >> report.md`

Reply:
also, unrelated but...


```suggestion
        if (node.url && fs.existsSync(node.url) && node.alt !== 'CML watermark') {
```

Reply:
> yes exactly, they're incompatible? The following will cause undefined behaviour, right?
> 
> 1. `echo '![pwned](https://examplewatermark.svg)' >> report.md`
> 2. `echo '![CML watermark](https://pwn.ed)' >> report.md`

Yes, our “watermarks” are utterly broken; we should be using a 128+ bits random string and in a consistent way. Feel free to open a separate issue. 👍🏼

Reply:
> _fs.existsSync(node.url)_

Why should we require linked files to exist beforehand? 🤔

Reply:
well at least check `!node.url.startsWith('http')`?

Reply:
Yes, we're adding remote links to the watcher list, but it doesn't quite matter for reasons other than performance.

Nitpick: what if users link to creatively named local files?

```
mkdir https:
touch https://file.com
```

Reply:
> _We definitely need to guard against existing URLs_

We already do; no link will be replaced unless a local file with the same path exists and can be successfully uploaded.

Reply:
Unless proven otherwise, the current implementation already behaves as you propose. Feel free to unresolve if there is any edge case where a remote URL is altered.

Reply:
ah right `https://file.com` will trigger `FileNotFound` ergo do-nothing, works for me :muk:

Reply:
Watermark issue moved to https://github.com/iterative/cml/issues/1076